.github/workflows/build_and_test_macos.yml

```diff
@@ -0,0 +1,48 @@
+name: Build and smoke test (MacOS)
+
+on:
+  pull_request:
+    branches: [master]
+  workflow_dispatch:
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        runs-on: [macos-latest]
```

Comment:
Only for `macos-latest`. This is ARM64.

https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners